pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -153,9 +153,15 @@ func (c *webhookAuthenticatorController) ensureKubeConfigSecret(ctx context.Cont
 		return nil, fmt.Errorf("failed to read kubeconfig template: %w", err)
 	}
 
+	// IPv6 addresses neeed to appear in brackets in URLs
+	apiserverHost := svc.Spec.ClusterIP
+	if strings.IndexByte(apiserverHost, ':') != -1 {
```

Comment:
isn't there a better way to do that? Like creating a url.URL ? Or some other IP -> url method?

Reply:
Could we build the URL used by authenticator-kubeconfig.yaml instead? I'm always concerned when there's environment variables named "IP" that are actually the URL host portion, since we risk creating the reverse bug: using a bracketed IPv6 when we don't want it.

We usually do something like this:
```go
apiEndpoint := fmt.Sprintf("https://%s/apis/oauth.openshift.io/v1/tokenreviews", net.JoinHostPort(svc.Spec.ClusterIP, 443))
```

Reply:
`JoinHostHost` does the right thing for IPv6, that's the only thing I've found in the stdlib.

Reply:
I would recommend using net.ParseIP(), then checking if To4(): 
```
		ip := net.ParseIP(address) // nil if not ip
		if ip.To4() != nil // you can write ipv4 in ipv6 format
                        fmt.Printf("%s", ip) 
                else 
                       fmt.Printf("[%s], ip) // ip.String() will be canonical, important for ipv6 addresses IMO
```

Reply:
@sanchezl I found that on the internets, but apparently, that may not work in some corner-cases where the IPv6 address only has the lower-bit values.

I modified the fix to include the solution by Stephen, seems standard enough.

Thank you both for suggestions :+1: